test/csx.coffee

```diff
@@ -786,3 +786,16 @@ test 'JSX fragments: fragment with component nodes', ->
       </Fragment>;
     };
   '''
+
+test '#5055: JSX expression indentation bug', ->
+  eqJS '''
+    <div>
+      {someCondition &&
+        <span />
+      }
+    </div>
+  ''', '''
+    <div>
+      {someCondition && <span />}
+    </div>;
+  '''
```

Comment:
How about a test that confirms that this works with a few other types of line continuators, besides just `&&`? And what if the next line is at the same indentation, and what if it’s normal code rather than a CSX tag?

Reply:
@GeoffreyBooth sure, added more tests with those variants